zeroize/src/lib.rs

```diff
@@ -471,7 +471,17 @@ where
         // object for at least `self.len()` elements of type `Z`.
         // `self.len()` is also not larger than an `isize`, because of the assertion above.
         // The memory of the slice should not wrap around the address space.
-        unsafe { volatile_set(self.as_mut_ptr(), Z::default(), self.len()) };
+        for z in self.iter_mut() {
+            *z = Z::default();
+        }
+        unsafe {
+            core::arch::asm!(
+                "/* {ptr} */",
+                ptr = in(reg) self.as_mut_ptr(),
+                options(nostack, readonly, preserves_flags),
+            );
```

Comment:
`asm!` is only stable for x86/x86-64,  ARM/AArch64, and RISC-V, so its usage needs to be gated for those platforms

Reply:
https://github.com/RustCrypto/utils/blob/f8f6f6e703a62dcbe0cc996c9b77bac2a9e9ad9f/zeroize/README.md?plain=1#L32

This guarantee needs to be updated if this change is merged.

Reply:
Elsewhere we feature-gate `asm`, so we could perhaps maintain that guarantee so long as the `asm` feature is off